### PR TITLE
Improving "quiet" login in Sanic exceptions

### PR DESCRIPTION
**Pull request by tomaszdrozdz**
Logic for setting quiet attribute for exception instance may be wrong,
or I do not understand intention behind it (if so please explain it to me).

What I mean by wrong:

!500 and !None

| quiet provided to decorator | quiet provided to init | quiet in exception instance ||
|-|-|-|-|
| True | False | True | But because I explicitly provided False when instantiating exception I expect quiet in this instance to be False |
| False | None | True | But because I explicitly provided False to decorator I expect quiet in this instance  to be False |

I can plot table with more examples like that.

So I propose a more straightforward logic for setting quiet attribute.
Please take a look, comment or If I am wrong explain.

### CODE REVIEW

File: sanic/exceptions.py [hunk 1]
```diff
@@ -38,10 +38,10 @@ def __init__(self, message, status_code=None, quiet=None):
             self.quiet = quiet
         elif hasattr(type(self), "quiet"):
             pass
-        elif status_code not in (None, 500):
-            self.quiet = True
-        else
+        elif status_code in (None, 500):
```
Comment (ahopkins) [inline sanic/exceptions.py:41]: Why this change?

I'd rather see it without the multiple conditionals.

```python
else:
    self.quiet = status_code not in (None, 500)
```

**Review by ahopkins — CHANGES_REQUESTED** [review]

File: sanic/exceptions.py (hunk 1 above)
Comment (tomaszdrozdz) [reply sanic/exceptions.py]: OK.

**Comment by tomaszdrozdz** [comment]
Are there any more obstacles, issues that can be introduced by this change ?
Can we merge it to master ?

File: sanic/exceptions.py [hunk 2]
```diff
@@ -11,9 +11,14 @@ def add_status_code(code, quiet=None):
 
     def class_decorator(cls):
         cls.status_code = code
-        if quiet or quiet is None and code != 500:
-            cls.quiet = True
+
+        if quiet is not None:
+            cls.quiet = quiet
+        else
```
Comment (AntonDnepr) [inline sanic/exceptions.py:17]: This and line 39 are triggering `SyntaxError: invalid syntax`, you missed `:`

**Comment by ahopkins** [comment]
Breaking changes need to be tabled until after LTS.

**Comment by ahopkins** [comment]
@tomaszdrozdz Are you able to make these changes?

**Comment by ahopkins** [comment]
I am closing this. If someone things there is something we still need here, please reopen a new PR.
